mockagent_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2020 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package jaeger
+
+import (
+	"context"
+	"net"
+	"sync"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"storj.io/monkit-jaeger/gen-go/agent"
+	"storj.io/monkit-jaeger/gen-go/jaeger"
+)
+
+// StartMockAgent starts a mock agent on a local udp port.
+func StartMockAgent(hostPort string) (*MockAgent, error) {
+	addr, err := net.ResolveUDPAddr("udp", hostPort)
+	if err != nil {
+		return nil, err
+	}
+	conn, err := net.ListenUDP(addr.Network(), addr)
+	if err != nil {
+		return nil, err
+	}
+	mock := &MockAgent{
+		batches: make([]*jaeger.Batch, 0),
+		mu:      &sync.Mutex{},
```

Comment:
you shouldn't need a pointer to a sync.Mutex. If you make it a non-pointer, you don't need to initialize it. Just make sure the struct is always being passed by pointer.